config/class-sync.php

```diff
@@ -63,7 +85,61 @@ public function filter_cron_schedules( $schedule ) {
 		return $schedule;
 	}
 
-	public function do_cron() {
+	public function get_blogs_to_sync() {
+		return $this->blogs_to_sync;
+	}
+
+	public function queue_sync_for_blog() {
+		// Queue the sync only from admin or CLI, to keep it out of frontend request path
+		if ( ! is_admin() && ! Context::is_wp_cli() ) {
+			return;
+		}
+		$blog_id = get_current_blog_id();
+
+		// To avoid performance issues, don't add if the count would surpass BLOGS_TO_SYNC_LIMIT.
+		// Can rely on the default cron schedules to sync the rest.
+		if ( count( $this->blogs_to_sync ) >= self::BLOGS_TO_SYNC_LIMIT ) {
+			return;
+		}
+
+		if ( ! in_array( $blog_id, $this->blogs_to_sync ) ) {
+			array_push( $this->blogs_to_sync, $blog_id );
+		}
+	}
+
+	public function run_sync_checks() {
+		if ( 0 === count( $this->blogs_to_sync ) ) {
+			return;
+		}
+
+		$needs_sync = false !== array_search( $this->original_blog_id, $this->blogs_to_sync );
+
+		if ( $needs_sync && ! $this->is_sync_ratelimited() ) {
+			if ( function_exists( 'fastcgi_finish_request' ) ) {
+				fastcgi_finish_request();
+			}
+
+			$this->put_site_details();
+		}
+
+		foreach ( $this->blogs_to_sync as $blog_id ) {
+			if ( $blog_id !== $this->original_blog_id ) {
+				switch_to_blog( $blog_id );
+
+				// Schedule a cron event to sync the blog data asap.
+				// Avoid syncing in this request as switch_to_blog() is not reliable for gaining the full state (plugins, etc).
+				if ( ! wp_next_scheduled( self::CRON_EVENT_NAME, [ 'is_faster_cron' => true ] ) ) {
+					wp_schedule_single_event( time() + 1, self::CRON_EVENT_NAME, [ 'is_faster_cron' => true ] );
+				}
+
+				restore_current_blog();
+			}
+		}
+		//clearing the sync queue after the run so that we don't sync the same blog twice
+		$this->blogs_to_sync = array();
```

Comment:
@WPprodigy I've also added this to clear the sync queue after we're done with it. If there is a slight possibility that the function might get triggered more than once then I want to be sure it's not sending double the notifications it needs.